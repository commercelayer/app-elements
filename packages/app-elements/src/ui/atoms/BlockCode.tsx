interface BlockCodeProps {
  /**
   * Object to render as pre-formatted block
   */
  json?: object
}

function BlockCode({ json, ...rest }: BlockCodeProps): JSX.Element {
  return (
    <div className='whitespace-pre font-mono' {...rest}>
      {json != null && Object.keys(json).length > 0 ? (
        <>{JSON.stringify(json, null, 2)}</>
      ) : (
        <>-</>
      )}
    </div>
  )
}

BlockCode.displayName = 'Badge'
export { BlockCode }
