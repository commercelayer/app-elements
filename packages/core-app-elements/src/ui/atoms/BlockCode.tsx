interface BlockCodeProps {
  /**
   * Object to render as pre-formatted block
   */
  json?: object
}

export function BlockCode({ json, ...rest }: BlockCodeProps): JSX.Element {
  return (
    <div className='whitespace-pre font-mono' {...rest}>
      {json != null && Object.keys(json).length > 0 ? (
        <>{JSON.stringify(json, null, 2)}</>
      ) : (
        <>-</>
      )}
    </div>
  )
}

export default BlockCode
