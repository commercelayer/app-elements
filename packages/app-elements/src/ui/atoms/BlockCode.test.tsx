import { render, type RenderResult } from '@testing-library/react'
import { BlockCode } from './BlockCode'

interface SetupProps {
  id: string
  json?: object
}

type SetupResult = RenderResult & {
  element: HTMLElement
}

const setup = ({ id, json }: SetupProps): SetupResult => {
  const utils = render(<BlockCode data-testid={id} json={json} />)
  const element = utils.getByTestId(id)
  return {
    element,
    ...utils
  }
}

describe('BlockCode', () => {
  test('Should be rendered', () => {
    const json = {
      foo: 'bar',
      hello: 'world'
    }
    const { element } = setup({
      id: 'my-code',
      json
    })
    expect(element).toBeInTheDocument()
    expect(JSON.parse(element.innerHTML)).toMatchObject(json)
    expect(element.innerHTML).toBe(`{\n  "foo": "bar",\n  "hello": "world"\n}`)
  })

  test('Should be a dash when no json is passed', () => {
    const { element } = setup({
      id: 'my-code',
      json: undefined
    })
    expect(element).toMatchSnapshot()
  })
})
