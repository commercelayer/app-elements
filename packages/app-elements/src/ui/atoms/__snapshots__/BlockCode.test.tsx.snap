// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlockCode > Should be a dash when no json is passed 1`] = `
<div
  class="whitespace-pre font-mono"
  data-testid="my-code"
>
  -
</div>
`;
